walletkit_client.go

```diff
@@ -737,28 +738,69 @@ func (m *walletKitClient) ListSweepsVerbose(ctx context.Context,
 	return result, nil
 }
 
+// BumpFeeOption customizes a BumpFee call.
+type BumpFeeOption func(*walletrpc.BumpFeeRequest)
+
+// WithImmediate is an option for enabling the immediate mode of BumpFee. The
+// sweeper will sweep this input without waiting for the next batch.
+func WithImmediate() BumpFeeOption {
+	return func(r *walletrpc.BumpFeeRequest) {
+		r.Immediate = true
+	}
+}
+
+// WithTargetConf is an option for setting the target_conf of BumpFee. If set,
+// the underlying fee estimator will use the target_conf to estimate the
+// starting fee rate for the fee function. The value of feeRate passed to
+// BumpFee will be ignored.
+func WithTargetConf(targetConf uint32) BumpFeeOption {
+	return func(r *walletrpc.BumpFeeRequest) {
+		r.TargetConf = targetConf
+
+		// We unset sat_per_vbyte, because if it is specified, it is
+		// used instead of target_conf.
+		r.SatPerVbyte = 0
```

Comment:
I think that should not be done here, but its already done in LND that it's not allowed to set both.

Reply:
I removed this code and added a check that if WithTargetConf is used, feeRate is 0. If both are specified, it is not clear, what the user wants and it is better to fail than to risk doing a wrong thing (e.g. overpay).